### PR TITLE
Add support for endpointaccessgateway for AWS

### PR DESCRIPTION
**Pull request by guruchai**

### CODE REVIEW

File: plugins/modules/env.py [hunk 1]
```diff
@@ -999,7 +1018,10 @@ def main():
             force=dict(required=False, type='bool', default=False),
             wait=dict(required=False, type='bool', default=True),
             delay=dict(required=False, type='int', aliases=['polling_delay'], default=15),
-            timeout=dict(required=False, type='int', aliases=['polling_timeout'], default=3600)
+            timeout=dict(required=False, type='int', aliases=['polling_timeout'], default=3600),
+            endpoint_access_subnets=dict(required=False, type='list', elements='str'),
+            endpoint_access_scheme=dict(required=False, type='str')
```
Comment (wmudge) [inline plugins/modules/env.py:1023]: add `choices=['PUBLIC', 'PRIVATE']` to `endpoint_access_scheme`.

File: plugins/modules/env.py [hunk 2]
```diff
@@ -248,6 +248,18 @@
     aliases:
       - s3_guard
       - s3_guard_table_name
+  endpoint_access_scheme:
```
Comment (wmudge) [inline plugins/modules/env.py:251]: add `choices` documentation entry, listing `PRIVATE` and `PUBLIC`

File: plugins/modules/env.py [hunk 3]
```diff
@@ -248,6 +248,18 @@
...
+    description:
+      - (AWS)The scheme for the workload endpoint gateway. PUBLIC creates an external endpoint that can be accessed over the Internet. 
+        Defaults to PRIVATE which restricts the traffic to be internal to the VPC / Vnet. Relevant in Private Networks.
+    type: str
+    required: False
+  endpoint_access_subnets:
+    description:
+      - (AWS) The subnets to use for endpoint access gateway.
```
Comment (wmudge) [inline plugins/modules/env.py:259]: Explicitly state these are subnet _IDs_

**Review by wmudge — CHANGES_REQUESTED** [review]
Minor changes

**Review by Chaffelson — APPROVED** [review]

**Pull request merged**